osu.Game/Rulesets/Scoring/ScoreStore.cs

```diff
@@ -141,7 +149,7 @@ private Replay createLegacyReplay(StreamReader reader)
                 ));
             }
 
-            return new Replay { Frames = frames };
+            return new Replay { Frames = frames.OrderBy(f => f.Time).ToList() };

```

Comment:
I have actually also looked into this issue before. I don't think this is the right solution- I think frames with negative offsets should be dropped. Copying what I said in the discord from a month back:

----

I'm trying to process .osr files, and i'm seeing that occasionally, there will be negative time offsets in the replay. taking a closer look, i find that the cursor doesn't seem to change direction, so intuitively the offending frame should have actually occured in between the two frames surrounding it, but despite that, it has a negative time offset. E.g. in the following, the x coordinate smoothly increases, but the middle frame has a negative time offset.

```8|129.1358|49.08057|0
8|133.194|42.44482|0
-5|139.9735|32.95264|0
16|145.082|28.51416|0
20|150.3815|26.59521|0```

What exactly does this mean? is my best bet simply to throw away these replay points (and fix the time offset of the frame afterwards), or is there a way to sensibly use the data from these points?

----

Pictorally the 5 frames in the original .osr sequence look like:

```
  5
  4
  3
 2
1
```

But if you sort by time it would become:

```
  5
  4
  2
 3
1
```

Reply:
We should probably check against the stable code to see how this is happening to make sure our approach is correct.

Reply:
This is basically what osu-stable does also:

```
        internal void ReadReplayData(string replayData)
        {
            if (replayData.Length > 0)
            {
                string[] replaylines = replayData.Split(',');

                bReplayFrame lastF;
                if (Replay.Count > 0)
                    lastF = Replay[Replay.Count - 1];
                else
                    lastF = new bReplayFrame(0, 0, 0, pButtonState.None);

                foreach (string replayline in replaylines)
                {
                    if (replayline.Length == 0)
                        continue;

                    string[] data = replayline.Split('|');

                    if (data.Length < 4)
                        continue;
                    if (data[0] == @"-12345")
                    {
                        Seed = int.Parse(data[3]);
                        continue;
                    }

                    pButtonState buttons = (pButtonState)Enum.Parse(typeof(pButtonState), data[3]);

                    bReplayFrame newF = new bReplayFrame(int.Parse(data[0]) + lastF.time,
                        float.Parse(data[1], GameBase.nfi),
                        float.Parse(data[2], GameBase.nfi),
                        buttons);
                    Replay.Add(newF);

                    lastF = newF;
                }
            }
        }
```

Reply:
i mean checking how they are being written as negatives in the first place (not sure i trust stable replay playback to be 100% perfect even today)